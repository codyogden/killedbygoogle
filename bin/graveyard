#!/usr/bin/env node
const slugify = require('slugify');
const fs = require('fs');

const data = require('../graveyard.json');

slugify.extend({
  '+': '-plus',
  '@': '-at',
});

data.map((item) => {
  const newItem = item;
  newItem.slug = slugify(item.name, {
    lower: true,
  });
  return newItem;
});

fs.writeFile('public/graveyard.json', JSON.stringify(data), (err) => {
  const { log } = console;
  if (err) throw err;
  log('Slugs added. JSON minified.');
});
fs.mkdir("public/story/", (err) => {
  if (err) throw err;
  fs.writeFile('public/story/index.html', fs.readFileSync("story/template.html"), (err) => {
    const { log } = console;
    if (err) throw err;
    log('Story built.');
  });
});
