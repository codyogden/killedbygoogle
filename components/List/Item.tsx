import React from 'react';
import dynamic from 'next/dynamic';
import {
  format,
  formatDistance,
  parseISO,
  formatDistanceToNow,
} from 'date-fns';

import { ProductWithSlug } from 'types/Product';
import Badge from 'components/Badge'; 

const DeathIdiom = dynamic(() => import('./LeadPhrase'), { ssr: false });

// Import Styled Components
import {
  AgeRange,
  ContentContainer,
  Description,
  Icon,
  IconContainer,
  ListItem,
} from './Item.atoms';

export default function Item(props: ProductWithSlug) {

  const isPast = () => {
    return new Date() > new Date(props.dateClose);
  };


  const dateOpen = parseISO(props.dateOpen);
  const dateClose = parseISO(props.dateClose);
  const relativeDate = formatDistanceToNow(dateClose);

  const getYears = () => {
    const duration = formatDistance(dateClose, dateOpen);

    if (!isPast()) {
      return ` It will be ${duration} old.`;
    }
    return ` It was ${duration} old.`;
  };

  const getIcon = () => {
    return isPast() ? (
      <Icon src='https://static.killedbygoogle.com/com/tombstone.svg' alt="Tombstone" />
    ) : (
      <Icon src='https://static.killedbygoogle.com/com/guillotine.svg' alt="Guillotine" />
    );
  };

  const ageRange = () => {
    const yearOpen = dateOpen.getFullYear();
    const yearClose = dateClose.getFullYear();
    if (!isPast()) {
      const monthClose = format(dateClose, 'LLLL');
      return (
        <AgeRange>
          <time dateTime={props.dateClose} title={`${props.dateClose}`}>
            {monthClose}
            <br />
            {yearClose}
          </time>
        </AgeRange>
      );
    }
    return (
      <AgeRange>
        <time dateTime={props.dateOpen} title={props.dateOpen}>
          {yearOpen}
        </time>
        {' - '}
        <time dateTime={props.dateClose} title={props.dateClose}>
          {yearClose}
        </time>
      </AgeRange>
    );
  };
  return (
    <ListItem>
      <IconContainer>
        {getIcon()}
        {ageRange()}
        <Badge content={props.type} />
      </IconContainer>
      <ContentContainer>
        <h2>
          <a href={props.link} target="_blank" rel="noopener noreferrer">
            {props.name}
          </a>
        </h2>
        <Description>
          {(isPast()) ? `Killed ${relativeDate} ago, ` : <DeathIdiom relativeDate={relativeDate} /> }
          {props.description}
          {getYears()}
        </Description>
      </ContentContainer>
    </ListItem>
  );
}

