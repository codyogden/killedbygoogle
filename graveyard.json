[
  {
    "dateClose": "2019-03-31",
    "dateOpen": "2015-05-28",
    "description": "Inbox by Gmail aimed to improve email through several key features.",
    "link": "https://en.wikipedia.org/wiki/Inbox_by_Gmail",
    "name": "Inbox by Gmail"
  },
  {
    "dateClose": "2012-10-05",
    "dateOpen": "2011-02-09",
    "description": "Sparrow was an email client for OS X and iOS. They acquired and then killed it.",
    "link": "https://en.wikipedia.org/wiki/Sparrow_(email_client)",
    "name": "Sparrow"
  },
  {
    "dateClose": "2006-12-01",
    "dateOpen": "2002-04-18",
    "description": "Google Answers was an online knowledge market.",
    "link": "https://en.wikipedia.org/wiki/Google_Answers",
    "name": "Google Answers"
  },
  {
    "dateClose": "2006-05-08",
    "dateOpen": "2003-11-10",
    "description": "Google Deskbar was a small inset window on the Windows toolbar and lets users perform searches without leaving desktop.",
    "link": "https://www.computerweekly.com/news/2240053391/Google-launches-Deskbar-software",
    "name": "Google Deskbar"
  },
  {
    "dateClose": "2006-10-10",
    "dateOpen": "2005-08-01",
    "description": "Writely was a Web-based word processor.",
    "link": "https://en.wikipedia.org/wiki/Writely",
    "name": "Writely"
  },
  {
    "dateClose": "2007-12-01",
    "dateOpen": "2004",
    "description": "Google Click-to-Call allowed a user to speak directly over the phone to businesses found in search results.",
    "link": "https://en.wikipedia.org/wiki/AdWords#Google_Click-to-Call",
    "name": "Google Click-to-Call"
  },
  {
    "dateClose": "2013-07-01",
    "dateOpen": "2005-10-07",
    "description": "Google Reader was an RSS/Atom feed aggregator.",
    "link": "https://en.wikipedia.org/wiki/Google_Reader",
    "name": "Google Reader"
  },
  {
    "dateClose": "2012-04-30",
    "dateOpen": "2009-09-01",
    "description": "Google Wave was an online communication and collaborative real-time editor tool.",
    "link": "https://en.wikipedia.org/wiki/Google_Wave",
    "name": "Google Wave"
  },
  {
    "dateClose": "2011-11-11",
    "dateOpen": "2008-08-10",
    "description": "Google Notebook allowed users to save and organize clips of information while conducting research online.",
    "link": "https://en.wikipedia.org/wiki/Google_Notebook",
    "name": "Google Notebook"
  },
  {
    "dateClose": "2018-04-30",
    "dateOpen": "2010-05-31",
    "description": "Encryped Search provided users with anonymous internet searching.",
    "link": "https://en.wikipedia.org/wiki/Google_Search#Dedicated_encrypted_search_page",
    "name": "Encrypted Search"
  },
  {
    "dateClose": "2018-08-20",
    "dateOpen": "2010-10-05",
    "description": "Google Goggles was used for searches based on pictures taken by handheld devices.",
    "link": "https://en.wikipedia.org/wiki/Google_Goggles",
    "name": "Google Goggles"
  },
  {
    "dateClose": "2013-11-01",
    "dateOpen": "2005-05-19",
    "description": "iGoogle was a customizable Ajax-based start page or personal web portal.",
    "link": "https://en.wikipedia.org/wiki/IGoogle",
    "name": "iGoogle"
  },
  {
    "dateClose": "2017-02-16",
    "dateOpen": "2016-05-16",
    "description": "Google Spaces was an app for group discussions and messaging.",
    "link": "https://en.wikipedia.org/wiki/Google_Spaces_(service)",
    "name": "Google Spaces"
  },
  {
    "dateClose": "2017-03-31",
    "dateOpen": "2008-06-23",
    "description": "Google Map Maker was a mapping and map editing service where users were able to draw features directly onto a map.",
    "link": "https://en.wikipedia.org/wiki/Google_Map_Maker",
    "name": "Google Map Maker"
  },
  {
    "dateClose": "2017-09-01",
    "dateOpen": "2007-03-01",
    "description": "Trendalyzer was a data trend viewing platform.",
    "link": "https://en.wikipedia.org/wiki/Trendalyzer",
    "name": "Trendalyzer"
  },
  {
    "dateClose": "2015-10-09",
    "dateOpen": "2002-10-15",
    "description": "Picasa was an image organizer and image viewer for organizing and editing digital photos.",
    "link": "https://en.wikipedia.org/wiki/Picasa",
    "name": "Picasa"
  },
  {
    "dateClose": "2016-01-15",
    "dateOpen": "2005-03-17",
    "description": "Google Code was a service that provided revision control, an issue tracker, and a wiki for code documentation.",
    "link": "https://en.wikipedia.org/wiki/Google_Developers#Google_Code",
    "name": "Google Code"
  },
  {
    "dateClose": "2016-07-01",
    "dateOpen": "2011-07-01",
    "description": "Google Swiffy was a web-based tool that converted SWF files to HTML5.",
    "link": "https://en.wikipedia.org/wiki/Google_Swiffy",
    "name": "Google Swiffy"
  },
  {
    "dateClose": "2016-11-04",
    "dateOpen": "2005-10-23",
    "description": "Panoramio was geo-located tagging, photo sharing mashup product.",
    "link": "https://en.wikipedia.org/wiki/Panoramio",
    "name": "Panoramio"
  },
  {
    "dateClose": "2015-04-20",
    "dateOpen": "2013-11-15",
    "description": "Google Helpouts was an online collaboration service where users to share their expertise through live video.",
    "link": "https://en.wikipedia.org/wiki/Google_Helpouts",
    "name": "Google Helpouts"
  },
  {
    "dateClose": "2015-06-30",
    "dateOpen": "2008-09-01",
    "description": "Google Moderator was a service that used crowdsourcing to rank user-submitted questions, suggestions and ideas.",
    "link": "https://en.wikipedia.org/wiki/Google_Moderator",
    "name": "Google Moderator"
  },
  {
    "dateClose": "2014-06-23",
    "dateOpen": "2007-06-28",
    "description": "Google Questions and Answers was a free knowledge market that allowed users to collaboratively find good answers to their questions.",
    "link": "https://en.wikipedia.org/wiki/Google_Questions_and_Answers",
    "name": "Google Questions and Answers"
  },
  {
    "dateClose": "2014-09-30",
    "dateOpen": "2004-01-24",
    "description": "Orkut was a social network designed to help users meet new and old friends and maintain existing relationships.",
    "link": "https://en.wikipedia.org/wiki/Orkut",
    "name": "Orkut"
  },
  {
    "dateClose": "2014-01-06",
    "dateOpen": "2013-09-19",
    "description": "Quickoffice was a productivity suite for mobile devices which allowed viewing, creating and editing documents, presentations and spreadsheets.",
    "link": "https://en.wikipedia.org/wiki/Quickoffice",
    "name": "Quickoffice"
  },
  {
    "dateClose": "2014-01-31",
    "dateOpen": "2005-01-01",
    "description": "Google Notifier alerted users to new messages in their Gmail account.",
    "link": "https://www.theverge.com/2014/1/16/5316748/google-quietly-kills-off-its-notifier-service-for-mail-and-calendar",
    "name": "Google Notifier"
  },
  {
    "dateClose": "2012-07-11",
    "dateOpen": "2005-09-14",
    "description": "Meebo was a browser-based instant messaging application which supported multiple IM services.",
    "link": "https://en.wikipedia.org/wiki/Meebo",
    "name": "Meebo"
  },
  {
    "dateClose": "2012-12-31",
    "dateOpen": "2007-12-06",
    "description": "Google Chart API was an interactive Web service that creates graphical charts from user-supplied data.",
    "link": "https://en.wikipedia.org/wiki/Google_Chart_API",
    "name": "Google Chart API"
  },
  {
    "dateClose": "2012-12-31",
    "dateOpen": "2007-12-06",
    "description": "Google Mini smaller version of the Google Search Appliance.",
    "link": "https://en.wikipedia.org/wiki/Google_Search_Appliance#Google_Mini",
    "name": "Google Mini"
  },
  {
    "dateClose": "2018-12-31",
    "dateOpen": "2002-01-01",
    "description": "Google Search Appliance was a rack-mounted device that provided document indexing functionality.",
    "link": "https://en.wikipedia.org/wiki/Google_Search_Appliance",
    "name": "Google Search Appliance"
  },
  {
    "dateClose": "2019-03-30",
    "dateOpen": "2009-12-01",
    "description": "Google URL Shortener, also known as goo.gl, is a URL shortening service.",
    "link": "https://en.wikipedia.org/wiki/Google_URL_Shortener",
    "name": "Google URL Shortener"
  },
  {
    "dateClose": "2011-12-15",
    "dateOpen": "2010-02-09",
    "description": "Google Buzz was a social networking, microblogging and messaging tool that integrated with Gmail.",
    "link": "https://en.wikipedia.org/wiki/Google_Buzz",
    "name": "Google Buzz"
  },
  {
    "dateClose": "2011-09-01",
    "dateOpen": "2008-01-01",
    "description": "Google Desktop allowed local searches of user's email messages, computer files, music, photos, chats, Web pages viewed.",
    "link": "https://en.wikipedia.org/wiki/Google_Desktop",
    "name": "Google Desktop"
  },
  {
    "dateClose": "2017-12-06",
    "dateOpen": "2010-12-06",
    "description": "Google Chrome Apps were a hosted or packaged web application that ran on the Google Chrome browser.",
    "link": "https://en.wikipedia.org/wiki/Google_Desktop",
    "name": "Google Chrome Apps"
  },
  {
    "dateClose": "2006-09-15",
    "dateOpen": "2002-01-01",
    "description": "Google Public Service search provided government, non-profits, and academic organizations search results without ads.",
    "link": "https://webmasters.googleblog.com/2006/09/for-those-wondering-about-public.html",
    "name": "Google Public Service Search"
  },
  {
    "dateClose": "2007-08-11",
    "dateOpen": "2006-01-07",
    "description": "Google Video Marketplace was service that included a store where videos could be bought and rented.",
    "link": "https://www.hexus.net/tech/news/software/4324-google-video-marketplace/",
    "name": "Google Video Marketplace"
  },
  {
    "dateClose": "2008-06-30",
    "dateOpen": "2006-06-08",
    "description": "Google Browser Sync was a Firefox extension that synced information like passwords and browsing history.",
    "link": "https://en.wikipedia.org/wiki/Google_Browser_Sync",
    "name": "Google Browser Sync"
  },
  {
    "dateClose": "2008-12-31",
    "dateOpen": "2008-07-08",
    "description": "Google Lively was a web-based virtual environment that provided a new way to access information.",
    "link": "https://en.wikipedia.org/wiki/Google_Lively",
    "name": "Google Lively"
  },
  {
    "dateClose": "2008-05-15",
    "dateOpen": "2002-08-01",
    "description": "Hello was a service by Picasa that let users share pictures \"like you're sitting side-by-side.\"",
    "link": "https://en.wikipedia.org/wiki/Picasa#Hello",
    "name": "Hello"
  },
  {
    "dateClose": "2008-11-24",
    "dateOpen": "2006-10-01",
    "description": "SearchMash was an experimental, non-branded search engine that Google used to be able to play around with new search technologies, concepts and interfaces.",
    "link": "https://techcrunch.com/2008/11/24/why-did-google-discontinue-searchmash/",
    "name": "SearchMash"
  },
  {
    "dateClose": "2009-02-28",
    "dateOpen": "2003-06-30",
    "description": "Dodgeball was a location-based social network where users texted their location to the service, and it notified them of friends and points of interest nearby.",
    "link": "https://en.wikipedia.org/wiki/Dodgeball_(service)",
    "name": "Dodgeball"
  },
  {
    "dateClose": "2009-07-31",
    "dateOpen": "2007-05-30",
    "description": "Google Mashup Editor was an online web mashup creation service with publishing, syntax highlighting, and debugging.",
    "link": "https://en.wikipedia.org/wiki/Google_Mashup_Editor",
    "name": "Google Mashup Editor"
  },
  {
    "dateClose": "2009-10-31",
    "dateOpen": "2005-05-01",
    "description": "Google Ride Finder was a service that used GPS data to pinpoint and map the location of taxis, limos, and shuttle vehicles available for hire in 10 U.S. metro areas.",
    "link": "https://searchenginewatch.com/sew/news/2062429/new-from-google-labs-google-ride-finder#",
    "name": "Google Ride Finder"
  },
  {
    "dateClose": "2012-01-01",
    "dateOpen": "2008-05-20",
    "description": "Google Health was a personal health information centralization service that provided users a merged health record from multiple sources.",
    "link": "https://en.wikipedia.org/wiki/Google_Health",
    "name": "Google Health"
  },
  {
    "dateClose": "2012-08-21",
    "dateOpen": "1999-07-01",
    "description": "Postini was an e-mail, Web security, and archiving service that filtered e-mail spam and malware (before it was delivered to a client's mail server), e-mail archiving.",
    "link": "https://en.wikipedia.org/wiki/Postini",
    "name": "Postini"
  },
  {
    "dateClose": "2010-02-19",
    "dateOpen": "1998-11-30",
    "description": "Marratech was a Swedish company that made software for e-meetings (e.g., web conferencing, videoconferencing).",
    "link": "https://en.wikipedia.org/wiki/Marratech",
    "name": "Marratech e-meetings"
  },
  {
    "dateClose": "2010-03-03",
    "dateOpen": "2008-11-20",
    "description": "SearchWiki was a Google Search feature which allowed logged-in users to annotate and re-order search results.",
    "link": "https://en.wikipedia.org/wiki/Google_SearchWiki",
    "name": "Google SearchWiki"
  },
  {
    "dateClose": "2010-11-12",
    "dateOpen": "2007-04-06",
    "description": "GOOG-411 (or Google Voice Local Search) was a telephone service that provided a speech-recognition-based business directory search.",
    "link": "https://en.wikipedia.org/wiki/GOOG-411",
    "name": "GOOG-411"
  },
  {
    "dateClose": "2010-12-17",
    "dateOpen": "2005-10-25",
    "description": "Google Base was a database provided by Google into which any user can add almost any type of content, such as text, images, and structured information.",
    "link": "https://en.wikipedia.org/wiki/Google_Base",
    "name": "Google Base"
  },
  {
    "dateClose": "2011-07-31",
    "dateOpen": "2002-05-20",
    "description": "Google Labs was a technology playground used by Google to demonstrate and test new projects.",
    "link": "https://en.wikipedia.org/wiki/Google_Labs",
    "name": "Google Labs"
  },
  {
    "dateClose": "2011-09-16",
    "dateOpen": "2009-10-05",
    "description": "Google PowerMeter was a software project of Google's philanthropic arm that helped consumers track their home electricity usage.",
    "link": "https://en.wikipedia.org/wiki/Google_PowerMeter",
    "name": "Google PowerMeter"
  },
  {
    "dateClose": "2018-10-08",
    "dateOpen": "2011-06-28",
    "description": "Google+ was an Internet-based social network.",
    "link": "https://en.wikipedia.org/wiki/Google%2B",
    "name": "Google+"
  },
  {
    "dateClose": "2014-02-07",
    "dateOpen": "2011-07-01",
    "description": "Google Schemer was a Google service for sharing and discovering things to do.",
    "link": "https://en.wikipedia.org/wiki/Google_Schemer",
    "name": "Google Schemer"
  },
  {
    "dateClose": "2013-08-09",
    "dateOpen": "2009-02-05",
    "description": "Google Latitude was a location-aware feature of Google Maps, a successor to an earlier SMS-based service Dodgeball.",
    "link": "https://en.wikipedia.org/wiki/Google_Latitude",
    "name": "Google Latitude"
  },
  {
    "dateClose": "2013-04-13",
    "dateOpen": "2007-01-01",
    "description": "Picnik was an online photo editing service that allowed users to edit, style, crop, and resize images.",
    "link": "https://en.wikipedia.org/wiki/Picnik",
    "name": "Picnik"
  },
  {
    "dateClose": "2016-11-01",
    "dateOpen": "2005-09-23",
    "description": "Google Showtimes was a standalone movie search result page.",
    "link": "https://techcrunch.com/2016/11/07/google-quietly-shutters-standalone-google-showtimes-movie-site/",
    "name": "Google Showtimes"
  },
  {
    "dateClose": "2014-01-31",
    "dateOpen": "2009-03-27",
    "description": "Bump! is a discontinued iOS and Android mobile app that enables smartphone users to transfer contact information, photos and files between devices.",
    "link": "https://en.wikipedia.org/wiki/Bump_(application)",
    "name": "Bump!"
  },
  {
    "dateClose": "2016-10-04",
    "dateOpen": "2010-01-05",
    "description": "Google Nexus was Google's line of flagship Android devices, superceded by Google Pixel.",
    "link": "https://en.wikipedia.org/wiki/Google_Nexus",
    "name": "Google Nexus"
  },
  {
    "dateClose": "2012-01-15",
    "dateOpen": "2006-02-01",
    "description": "Jaiku was a social networking, micro-blogging and lifestreaming service comparable to Twitter.",
    "link": "https://en.wikipedia.org/wiki/Jaiku",
    "name": "Jaiku"
  },
  {
    "dateClose": "2012-05-01",
    "dateOpen": "2008-07-23",
    "description": "Knol was a Google project that aimed to include user-written articles on a range of topics.",
    "link": "https://en.wikipedia.org/wiki/Knol",
    "name": "Knol"
  },
  {
    "dateClose": "2015-01-01",
    "dateOpen": "2014-05-01",
    "description": "Google Play was an edition of an Android phone that Google sold",
    "link": "https://arstechnica.com/gadgets/2015/01/dont-cry-for-the-google-play-edition-program-it-was-already-dead/",
    "name": "Google Play"
  },
  {
    "dateClose": "2016-03-01",
    "dateOpen": "2015-01-01",
    "description": "Google Compare allowed consumers to compare several offers ranging from insurance, mortgage, and credit cards.",
    "link": "https://www.gadgetsnow.com/year-in-review-2016/Google-Compare/yearendershow2016/55617272.cms",
    "name": "Google Compare"
  },
  {
    "dateClose": "2011-04-03",
    "dateOpen": "2009-11-12",
    "description": "Gizmo5 was a VOIP communications network and a proprietary freeware soft phone for that network.",
    "link": "https://en.wikipedia.org/wiki/Gizmo5",
    "name": "Gizmo5"
  },
  {
    "dateClose": "2012-01-15",
    "dateOpen": "2006-10-05",
    "description": "Google Code Search was a free beta product from Google which debuted in Google Labs on October 5, 2006, allowing web users to search for open-source code on the Internet.",
    "link": "https://en.wikipedia.org/wiki/Google_Code_Search",
    "name": "Google Code Search"
  },
  {
    "dateClose": "2014-02-17",
    "dateOpen": "2013-07-13",
    "description": "SlickLogin was an Israeli start-up company which developed sound-based password alternatives, was acquired by Google and hasn't released anything since.",
    "link": "https://en.wikipedia.org/wiki/SlickLogin",
    "name": "SlickLogin"
  },
  {
    "dateClose": "2008-08-31",
    "dateOpen": "2006-07-13",
    "description": "Google Page Creator was a website creation and hosting service by Google. It was a tool for basic website design, requiring no HTML knowledge.",
    "link": "https://en.wikipedia.org/wiki/Google_Page_Creator",
    "name": "Google Page Creator"
  },
  {
    "dateClose": "2017-12-15",
    "dateOpen": "2014-12-01",
    "description": "Project Tango was an API for augmented reality apps that was killed and replaced by ARCore",
    "link": "https://en.wikipedia.org/wiki/Tango_(platform)",
    "name": "Project Tango"
  },
  {
    "dateClose": "2019-06-01",
    "dateOpen": "2014-10-01",
    "description": "Fabric was a platform that helped mobile teams build better apps, understand their users, and grow their business.",
    "link": "https://get.fabric.io/roadmap",
    "name": "Fabric"
  },
  {
    "dateClose": "2013-07-01",
    "dateOpen": "2012-06-27",
    "description": "Nexus Q is a discontinued digital media player developed by Google.",
    "link": "https://en.wikipedia.org/wiki/Nexus_Q",
    "name": "Nexus Q"
  },
  {
    "dateClose": "2015-08-04",
    "dateOpen": "2011-08-16",
    "description": "Google Catalogs was a shopping application that delivered the virual catalogs of large retailers to users.",
    "link": "https://en.wikipedia.org/wiki/Google_Catalogs",
    "name": "Google Catalogs"
  },
  {
    "dateClose": "2011-04-08",
    "dateOpen": "2010-11-15",
    "description": "Google Hotpot was a local recommendation engine that allowed people to rate restaurants, hotels, etc. and share them with friends.",
    "link": "https://mashable.com/2011/04/08/google-hotpot-goes-cold/",
    "name": "Google Hotpot"
  },
  {
    "dateClose": "2011-09-02",
    "dateOpen": "2009-09-23",
    "description": "A browser sidebar tool that allowed users to contribute information to any web page.",
    "link": "https://en.wikipedia.org/wiki/Google_Sidewiki",
    "name": "Google Sidewiki"
  },
  {
    "dateClose": "2012-07-26",
    "dateOpen": "2008-12-22",
    "description": "AdSense for Feeds was an RSS- based extension of the AdSense system that allowed to advertise on RSS Feeds. It was a spin-off of the half dormant half dead Feedburner",
    "link": "http://adsenseforfeeds.blogspot.com/",
    "name": "AdSense for Feeds"
  },
  {
    "dateClose": "2011-09-02",
    "dateOpen": "2009-03-14",
    "description": "Aardvark was a social search service that connected users live with friends or friends-of-friends who were able to answer their questions.",
    "link": "https://en.wikipedia.org/wiki/Aardvark_(search_engine)",
    "name": "Aardvark"
  },
  {
    "dateClose": "2009-03-30",
    "dateOpen": "2007-09",
    "description": "Google Shared Stuff was a web page sharing system that allowed users to bookmark pages and share it.",
    "link": "https://www.google.com/s2/sharing/stuff",
    "name": "Google Shared Stuff"
  },
  {
    "dateClose": "2007",
    "dateOpen": "2001",
    "description": "Google Zeitgeist was a weekly, monthly, and yearly snapshot in time of what people were searching for on Google all over the world.",
    "link": "https://www.lifewire.com/google-zeitgeist-3481903",
    "name": "Zeitgeist"
  },
  {
    "dateClose": "2015-10-02",
    "dateOpen": "2010-01",
    "description": "Google X was an American semi-secret research & development organization which now operates as a subsidiary of Alphabet Inc.",
    "link": "https://en.wikipedia.org/wiki/X_(company)",
    "name": "Google X"
  },
  {
    "dateClose": "2010-07-18",
    "dateOpen": "2010-01-05",
    "description": "The Nexus One (codenamed HTC Passion) is an Android smartphone designed and manufactured by HTC as Google's first Nexus smartphone.",
    "link": "https://en.wikipedia.org/wiki/Nexus_One",
    "name": "Google Nexus One"
  },
  {
    "dateClose": "2016-09-2",
    "dateOpen": "2013-10-29",
    "description": "Project Ara was a modular smartphone project under development by Google.",
    "link": "https://en.wikipedia.org/wiki/Project_Ara",
    "name": "Project Ara"
  },
  {
    "dateClose": "2013-05-15",
    "dateOpen": "2005-08-22",
    "description": "Often remembered as 'Gchat', Google Talk was a messaging service for both text and voice using XMPP. Google later replaced Google Talk with Hangouts.",
    "link": "https://en.wikipedia.org/wiki/Google_Talk",
    "name": "Google Talk"
  },
  {
    "dateClose": "2015-03-15",
    "dateOpen": "2013-04-30",
    "description": "BebaPay was a form of electronic ticketing platform in Nairobi, Kenya.",
    "link": "https://en.wikipedia.org/wiki/BebaPay",
    "name": "BebaPay"
  },
  {
    "dateClose": "2011-09-01",
    "dateOpen": "2009-09-14",
    "description": "Google Fast Flip was an online news aggregator, something of a high tech microfiche.",
    "link": "https://en.wikipedia.org/wiki/Google_Fast_Flip",
    "name": "Google Fast Flip"
  },
   {
    "dateClose": "2008-01-20",
    "dateOpen": "2005-05-04",
    "description": "Google Web Accelerator was a web accelerator produced by Google that increased load speed of web pages.",
    "link": "https://en.wikipedia.org/wiki/Google_Web_Accelerator",
    "name": "Google Web Accelerator"
  },
  {
    "dateClose": "2018-10-11",
    "dateOpen": "2018-02-01",
    "description": "Reply was an app which lets users to insert Smart Replies (pre-defined replies) into conversations on messaging apps on their phone such as Facebook Messenger, Slack and Hangouts",
    "link": "https://en.wikipedia.org/wiki/Reply_(Google)",
    "name": "Reply app"
  },
  {
    "dateClose": "2011-09-05",
    "dateOpen": "2009-06-03",
    "description": "Google Squared was an information extraction and relationship extraction product from Google.",
    "link": "https://en.wikipedia.org/wiki/Google_Squared",
    "name": "Google Squared"
  },
  {
    "dateClose": "2014-02-21",
    "dateOpen": "2012-11-30",
    "description": "BufferBox Inc. was a Canadian startup, acquired by Google, that provided consumers the convenience of picking up their online purchases 24/7.",
    "link": "https://en.wikipedia.org/wiki/BufferBox",
    "name": "BufferBox"
  }
]
