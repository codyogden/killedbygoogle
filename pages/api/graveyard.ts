import { NextApiRequest, NextApiResponse } from 'next';
import slugify from 'slugify';
import { Product } from 'types/Product';

export default async function handler(req: NextApiRequest, res: NextApiResponse) {
    const data = await require('../../graveyard.json');
    slugify.extend({
        '+': '-plus',
        '@': '-at',
    });
    res.setHeader('Cache-Control', 'max-age=3600, public');
    return res.status(200).json(data.map((item: Product) => ({
        ...item,
        slug: slugify(item.name, {
            lower: true,
        }),
    })));
}
