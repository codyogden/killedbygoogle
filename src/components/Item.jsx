import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { format, formatDistance } from 'date-fns';

// Import Styled Components
import {
  AgeRange,
  ContentContainer,
  Description,
  Icon,
  IconContainer,
  ListItem,
} from './Item.atoms';

const eolIdiom = () => {
  const items = [
    'Sentenced to death',
    '"Off with their heads!"',
    'Kicking the bucket',
    'Dead as a doorknob',
    'Done for',
    'Expires',
    'Bites the big one',
    'Off to the glue factory',
    'Another one bites the dust',
  ];
  return items[Math.floor(Math.random() * items.length)];
};

export default class Item extends Component {
  getIcon() {
    return (this.isPast()) ? <Icon src="assets/tombstone.svg" alt="Tombstone" /> : <Icon src="assets/guillotine.svg" alt="Guillotine" />;
  }

  getYears() {
    const { dateClose, dateOpen } = this.props;
    const duration = formatDistance(dateClose, dateOpen);

    return (` It was ${duration} old.`);
  }

  isPast() {
    const { dateClose } = this.props;
    return new Date() > new Date(dateClose);
  }

  timePhrase() {
    const { dateClose } = this.props;
    const relativeDate = formatDistance(new Date(dateClose), new Date());
    const exactDate = format(new Date(dateClose), 'MMMM yyyy');
    const yearFromNow = new Date().setFullYear(new Date().getFullYear() + 1);

    if (!this.isPast()) {
      if (new Date(dateClose) < yearFromNow) {
        return (
          <span>
            {`${eolIdiom()} in ${relativeDate}, `}
          </span>
        );
      }
      return (
        <span>
          {`${eolIdiom()} in ${exactDate}, `}
        </span>
      );
    }
    return (
      <span>
        {`Killed ${relativeDate} ago, `}
      </span>
    );
  }

  ageRange(grave) {
    if (!this.isPast()) {
      const date = new Date(grave.dateClose);
      return (
        <AgeRange>
          {date.toLocaleDateString('en-US', { month: 'long' })}
          <br />
          {date.toLocaleDateString('en-US', { year: 'numeric' })}
        </AgeRange>
      );
    }
    return (
      <AgeRange>
        {new Date(grave.dateOpen).getFullYear()}
        {' - '}
        {new Date(grave.dateClose).getFullYear()}
      </AgeRange>
    );
  }

  render() {
    const { ...grave } = this.props;
    return (
      <ListItem>
        <IconContainer>
          {this.getIcon()}
          {this.ageRange(grave)}
        </IconContainer>
        <ContentContainer>
          <h2><a href={grave.link} target="_blank" rel="noopener noreferrer">{grave.name}</a></h2>
          <Description>
            {this.timePhrase()}
            {grave.description}
            {this.getYears()}
          </Description>
        </ContentContainer>
      </ListItem>
    );
  }
}

Item.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  dateClose: PropTypes.string.isRequired,
  dateOpen: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};
