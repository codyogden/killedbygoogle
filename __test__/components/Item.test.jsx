import React from 'react';
import { shallow, mount } from 'enzyme';
import { willDieInFuture, diedInPast } from '../fixtures/graveyard.jsx';
import Item from '../../src/components/Item';

describe('Items with death dates (dateClose) in the past', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<Item {...diedInPast[0]} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render Icon image with tombstone', () => {
        const wrapper = mount(<Item {...diedInPast[1]} />);
        expect(
            wrapper.find("IconContainer").containsMatchingElement(
            <img src="assets/tombstone.svg" alt="Tombstone" />
            )
        ).toBeTruthy();
    });
});

describe('Items with death dates (dateClose) in the future', () => {
    test('should render Icon image with guillotine', () => {
        const wrapper = mount(<Item {...willDieInFuture[0]} />);    
        expect(
            wrapper.find("IconContainer").containsMatchingElement(
            <img src="assets/guillotine.svg" alt="Guillotine" />
            )
        ).toBeTruthy();
    });
});