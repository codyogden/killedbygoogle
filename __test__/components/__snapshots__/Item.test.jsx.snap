// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items with death dates (dateClose) in the past should render correctly 1`] = `
<ListItem>
  <IconContainer>
    <Icon
      alt="Tombstone"
      src="assets/tombstone.svg"
    />
    <AgeRange>
      2014
       - 
      2017
    </AgeRange>
  </IconContainer>
  <ContentContainer>
    <h2>
      <a
        href="https://en.wikipedia.org/wiki/Nexus_One"
      >
        Google Nexus One
      </a>
    </h2>
    <Description>
      <span>
        Killed about 1 year ago, 
      </span>
      The Nexus One (codenamed HTC Passion) is an Android smartphone designed and manufactured by HTC as Google's first Nexus smartphone.
       It was 3 years old.
    </Description>
  </ContentContainer>
</ListItem>
`;
